src/test/fetch.test.js

```diff
@@ -0,0 +1,77 @@
+import * as auth from '../js/api/auth';
+
+const TEST_ID = 1;
+const TEST_BAD_ID = 'banana';
+const TEST_STRING = 'Test';
+const TEST_TOKEN = 'Token';
+const TEST_PROFILE = {
+  id: TEST_ID,
+  name: TEST_STRING,
+  accessToken: TEST_TOKEN,
+};
+
+class LocalStorageMock {
+  constructor() {
+    this.store = {};
+  }
+
+  clear() {
+    this.store = {};
+  }
+
+  getItem(key) {
+    return this.store[key] || null;
+  }
+
+  setItem(key, value) {
+    this.store[key] = value.toString();
+  }
+
+  removeItem(key) {
+    delete this.store[key];
+  }
+}
+
+globalThis.localStorage = new LocalStorageMock();
```

Comment:
```suggestion
beforeAll(() => {
  globalThis.localStorage = new LocalStorageMock();
 });
```
Consider moving the mock assignment to a beforeAll or beforeEach hook: https://jestjs.io/docs/setup-teardown